src/Microsoft.Graph.Core/Authentication/TokenCredentials/IntegratedWindowsTokenCredential.cs

```diff
@@ -0,0 +1,155 @@
+        {   AuthConstants.Tenants.Common,
+            AuthConstants.Tenants.Consumers,
+            AuthConstants.Tenants.Organizations
+        };
+
+        private readonly IPublicClientApplication _clientApplication;
+
+        /// <summary>
+        /// Creates a new IntegratedWindowsTokenCredential which will authenticate users with the specified application.
+        /// </summary>
+        /// <param name="publicClientApplication">A <see cref="IPublicClientApplication"/> to pass to <see cref="IntegratedWindowsTokenCredential"/> for authentication.</param>
+        public IntegratedWindowsTokenCredential(IPublicClientApplication publicClientApplication)
+        {
+            _clientApplication = publicClientApplication ?? throw new AuthenticationException(
+                new Error
+                {
+                    Code = ErrorConstants.Codes.InvalidRequest,
+                    Message = string.Format(ErrorConstants.Messages.NullValue, nameof(publicClientApplication))
+                });
+        }
+
+        /// <summary>
+        /// Gets an <see cref="AccessToken"/> using the provided <see cref="TokenRequestContext"/> in a synchronous fashion
+        /// </summary>
+        /// <param name="requestContext">The <see cref="TokenRequestContext"/> for the request</param>
+        /// <param name="cancellationToken">The <see cref="CancellationToken"/> for the request</param>
+        /// <returns>An <see cref="AccessToken"/> to make requests with</returns>
+        public override AccessToken GetToken(TokenRequestContext requestContext, CancellationToken cancellationToken)
+        {
+            return GetTokenAsync(requestContext,cancellationToken).GetAwaiter().GetResult();
+        }
+
+        /// <summary>
+        /// Gets an <see cref="AccessToken"/> using the provided <see cref="TokenRequestContext"/> in an asynchronous fashion
+        /// </summary>
+        /// <param name="requestContext">The <see cref="TokenRequestContext"/> for the request</param>
+        /// <param name="cancellationToken">The <see cref="CancellationToken"/> for the request</param>
+        /// <returns>An <see cref="AccessToken"/> to make requests with</returns>
+        public override async ValueTask<AccessToken> GetTokenAsync(TokenRequestContext requestContext, CancellationToken cancellationToken)
+        {
+            AuthenticationResult result;
+            try
+            {
+                // Try to get the account from the cache
+                IEnumerable<IAccount> accounts = await _clientApplication.GetAccountsAsync();
+                IAccount account = accounts.FirstOrDefault();
+
+                // Try to get the token silently
+                AcquireTokenSilentParameterBuilder tokenSilentBuilder =
+                    _clientApplication.AcquireTokenSilent(requestContext.Scopes, account);
+
+                if (!ContainsWellKnownTenantName(_clientApplication.Authority))
```

Comment:
I don't understand this check. 

If the authority is one of the well known endpoints, the _clientApplication.Authority is used (I assume it is used by `tokenSilentBuilder`). If it isn't, as in `login.microsoftonline.com/{tenantId}`, then the same _clientApplication.Authority is set on the `tokenSilentBuilder`. What is the difference of setting the authority on the `_clientApplication` versus on `tokenSilentBuilder`?

ContainsWellKnownTenantName seems to be more of a check on the wellformedness of the authority URI.

Reply:
You are correct. 
Since the user will set the `Authority` at the client level there is no need to do this. This should ideally be use if a well known Authority name is used at the top level and one wishes to narrow down the tenant name when fetching the token silently.

Reply:
I have actioned on this and removed the check.